test/arrayimpl.py

```diff
@@ -526,19 +512,23 @@ def as_raw(self):
             return self.array
 
 
-def subTest(case, skip=(), skiptypecode=()):
+def loop(*args):
     for array in ArrayBackends:
-        if array.backend == skip:
-            continue
-        if array.backend in skip:
-            continue
+        loop.array = array
```

Comment:
Not a big fan of attaching attributes to a function... If `loop` has to have attributes, can't we just turn it into a class that has `__call__`?

Reply:
I use function attributes sparingly, as a way to track and hide things that would otherwise have to be stored in global variables. At this point, I do not want to complicate the implementation of all this silly extra testing support. If mpi4py would use a more advanced testing framework, then all this mess would probably not be there. But I do not want to depend on stuff outside Python's stdlib.